tests/jballerina-unit-test/src/test/resources/test-src/javainterop/ballerina_types_as_interop_types.bal

```diff
@@ -226,6 +233,10 @@ public function getArrayValueFromMapWhichThrowsCheckedException(string key, map<
     'class:"org/ballerinalang/nativeimpl/jvm/tests/StaticMethods"
 } external;
 
+public function acceptIntErrorUnionReturnWhichThrowsCheckedException(int flag) returns int|error = @java:Method {
```

Comment:
Shall we have a test case with distinct errors as well?

Reply:
Added with https://github.com/ballerina-platform/ballerina-lang/pull/33876/commits/f19a60666316fc0e9477b452977692e3e34be8da